File: Task3/task3.rs
use std::io;

fn main() {
    println!("Введите количество чисел N:");
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    println!("Введите {} чисел через пробел (или по одному на строку):", n);
    let mut numbers = Vec::new();
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    
    for word in line.split_whitespace() {
        numbers.push(word.to_string());
    }

    // Если ввели меньше чисел, чем N, добираем из оставшихся строк
    while numbers.len() < n {
        let mut extra = String::new();
        io::stdin().read_line(&mut extra).unwrap();
        for word in extra.split_whitespace() {
            numbers.push(word.to_string());
        }
    }

    let count = numbers.iter().take(n).filter(|num| num.len() % 2 == 0).count();
    println!("Результат: {}", count);
}

File: Task3/task3.kt
fun main() {

    System.setOut(java.io.PrintStream(System.out, true, "UTF-8"))
    
    print("Введите количество чисел N: ")
    val n = readln().toInt()

    println("Введите $n чисел через пробел:")
    val input = readln().trim().split("\\s+".toRegex())
    var count = 0

    for (i in 0 until n) {
        val num = if (i < input.size) input[i] else ""
        if (num.length % 2 == 0) count++
    }

    println("Результат: $count")
}